specification/fist/resource-manager/Microsoft.IoTFirmwareDefense/preview/2023-02-08-preview/firmwares.json

```diff
@@ -0,0 +1,1983 @@
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The uri to fetch the next page of asset."
+        }
+      }
+    },
+    "BinaryHardeningFeatures": {
+      "type": "object",
+      "properties": {
+        "nx": {
+          "description": "NX flag.",
+          "type": "string",
+          "enum": [
+            "True",
+            "False"
+          ],
+          "x-ms-enum": {
+            "name": "nxFlag",
+            "modelAsString": true
+          }
+        },
+        "pie": {
+          "description": "PIE flag.",
+          "type": "string",
+          "enum": [
+            "True",
+            "False"
+          ],
+          "x-ms-enum": {
+            "name": "pieFlag",
+            "modelAsString": true
+          }
+        },
+        "relro": {
+          "description": "RELRO flag.",
+          "type": "string",
+          "enum": [
+            "True",
+            "False"
+          ],
+          "x-ms-enum": {
+            "name": "relroFlag",
+            "modelAsString": true
+          }
+        },
+        "canary": {
+          "description": "Canary flag.",
+          "type": "string",
+          "enum": [
+            "True",
+            "False"
+          ],
+          "x-ms-enum": {
+            "name": "canaryFlag",
+            "modelAsString": true
+          }
+        },
+        "stripped": {
+          "description": "Stripped flag.",
+          "type": "string",
+          "enum": [
+            "True",
+            "False"
+          ],
+          "x-ms-enum": {
+            "name": "strippedFlag",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "Binary hardening features."
+    },
+    "BinaryHardening": {
+      "type": "object",
+      "properties": {
+        "binaryHardeningId": {
+          "description": "ID for the binary hardening result.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "features": {
+          "title": "BinaryHardeningFeatures",
+          "description": "Binary hardening features.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/BinaryHardeningFeatures"
+        },
+        "architecture": {
+          "description": "The architecture of the uploaded firmware.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "path": {
+          "description": "path for binary hardening.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "class": {
+          "description": "class for binary hardening.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "runpath": {
+          "description": "The runpath of the uploaded firmware.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "rpath": {
+          "description": "The rpath of the uploaded firmware.",
+          "type": "string",
+          "x-nullable": true
+        }
```

Comment:
What's an 'rpath'? Could it be named something else?

Reply:
It's specific to our security analysis, it's a known term in the firmware industry